core/src/main/scala/caliban/wrappers/Wrapper.scala

```diff
@@ -175,46 +175,47 @@ object Wrapper {
     )
   ] =
     if (wrappers.isEmpty) emptyWrappers
-    else {
-      val o = ListBuffer.empty[OverallWrapper[R]]
-      val p = ListBuffer.empty[ParsingWrapper[R]]
-      val v = ListBuffer.empty[ValidationWrapper[R]]
-      val e = ListBuffer.empty[ExecutionWrapper[R]]
-      val f = ListBuffer.empty[FieldWrapper[R]]
-      val i = ListBuffer.empty[IntrospectionWrapper[R]]
-
-      def loop(wrapper: Wrapper[R]): Option[URIO[R, Unit]] = wrapper match {
-        case wrapper: OverallWrapper[R]       => o append wrapper; None
-        case wrapper: ParsingWrapper[R]       => p append wrapper; None
-        case wrapper: ValidationWrapper[R]    => v append wrapper; None
-        case wrapper: ExecutionWrapper[R]     => e append wrapper; None
-        case wrapper: FieldWrapper[R]         => f append wrapper; None
-        case wrapper: IntrospectionWrapper[R] => i append wrapper; None
-        case SuspendedWrapper(f)              => loop(f())
-        case CombinedWrapper(wrappers)        =>
-          wrappers.flatMap(loop) match {
-            case Nil => None
-            case fs  => Some(ZIO.collectAllDiscard(fs))
-          }
-        case EffectfulWrapper(wrapper)        =>
-          Some(wrapper.flatMap {
-            loop(_) match {
-              case None    => Exit.unit
-              case Some(w) => w
+    else
+      ZIO.suspendSucceed {
```

Comment:
Added this here in order to align with https://github.com/ghostdogpr/caliban/pull/2338/files#r1673288713. We don't need this method to be referentially transparent because it's executed on each request, but better to have it referentially transparent for consistency